### PR TITLE
Add Private Relay test

### LINKED ISSUE #1554: Email Relay Test: Add custom dimensions in GA
Child of #1535

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
- Adds private relay callout to signed in breaches dashboard
  - "View" pings for the private relay CTA should only fire when users who have not already signed up scroll the CTA into view.
  - "Engage" pings should fire on click.
- Adds new private relay landing page
  - Signed in users who haven't signed up should see a "Join the Waitlist" CTA
  - Signed in users who _have_ signed up should see "You're on the list!"
  - "View" pings for the "Join the Waitlist" button should only fire when a signed in user, who hasn't signed up, scrolls the button into view.
  - Signed out users should be redirected to the /oauth flow.

- Adds WIP/placeholder routes and controllers

Fixes #1554 , #1553

### CODE REVIEW

File: public/js/monitor.js [hunk 1]
```diff
@@ -306,6 +306,67 @@ function addBentoObserver(){
...
+      try {
+        await fetch(relayEndpoint, {
+          headers: {
+            "Content-Type": "application/json; charset=utf-8",
+          },
+          mode: "same-origin",
+          method: "POST",
+          body: JSON.stringify({"emailToAdd": "add-user-email"}),
+        })
+        .then(async(data) => {
```
Comment (groovecoder) [inline public/js/monitor.js]: Should we use `then` when we're using `await`?

File: public/js/monitor.js [hunk 2]
```diff
@@ -306,6 +306,67 @@ function addBentoObserver(){
...
+        });
+      } catch(e) {
+        // we need error messaging
+      }
+    });
+  });
+
+  const privateRelayCtas = document.querySelectorAll(".private-relay-cta");
+
+  if (privateRelayCtas.length > 0) {
```
Comment (groovecoder) [inline public/js/monitor.js]: Can we refactor this into helper `function` like `addPrivateRelayCtaObservers` ? And then add an early `return;` if there are none.

```js

function addPrivateRelayCtaObservers() {
  const privateRelayCtas = document.querySelectorAll(".private-relay-cta");
  if (privateRelayCtas.length <= 0) {
    return;
  }

  const availableIntersectionObserver = ("IntersectionObserver" in window);
  ...
}
```

File: public/js/monitor.js [hunk 3]
```diff
@@ -306,6 +306,67 @@ function addBentoObserver(){
     });
   });
 
+  document.querySelectorAll(".relay-sign-up-btn").forEach(btn => {
```
Comment (groovecoder) [inline public/js/monitor.js]: Refactor into a helper function `addRelaySignUpBtnListeners`?

File: views/private-relay.hbs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      <h2 class="relay-headline txt-cntr txt-white">Keep your personal email private.</h2>
+      <h3 class="relay-subhead txt-cntr txt-light-grey">The best way to protect your personal info is to never give it out.</h3>
+    </div>
+
+    <section class="relay-info toggle-parent active">
+      <button class="toggle svg-wrap article-toggle relay-toggle-button" aria-label="show-content">
+        <span class="relay-info-subhead bold ff-Met txt-white">What is email masking?</span>
+        <svg class="toggle-down" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path class="relay-toggle" fill="#ffffff" d="M8 12a1 1 0 0 1-.707-.293l-5-5a1 1 0 0 1 1.414-1.414L8 9.586l4.293-4.293a1 1 0 0 1 1.414 1.414l-5 5A1 1 0 0 1 8 12z"></path></svg>
+      </button>
+      <p class="relay-info-body txt-light-grey toggle-child">Most apps require an email address. When you give out your real email address, it can be picked up by hackers who use it to take over your online accounts, or trackers who sell or trade it online to make money. When you use email masking, you give apps a random, anonymous email address to keep your real email address private.</p>
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply views/private-relay.hbs]: Oops - I've been pulling it from the invision spec. I will update with the latest!

File: public/js/monitor.js (hunk 1 above)
Comment (lesleyjanenorton) [reply public/js/monitor.js]: It works but what I just pushed up should be easier to read.

**Comment by groovecoder** [comment]
Okay, the controller code has been wired to the DB to save subscribers on the wait-list, and wired to the view to show a user's real wait-list status. This is ready for final review.

File: public/img/svg/email-relay-touts.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...0,0,0-2.93-4.62h-.09l.09.09-52.37,50.11,2.35,2.47,50.32,52.77a10.75,10.75,0,0,0,3.17-7.63V59.37a10.3,10.3,0,0,0-.54-3.31Z" transform="translate(-0.12 -0.02)" fill="#ff298a"/><path d="M129.8,102.63l18.94,17.86,19.68-18.83,52.37-50.12-.09-.08a10.38,10.38,0,0,0-7.19-3H84a10.55,10.55,0,0,0-7.62,3.34l-.09.1Z" transform="translate(-0.12 -0.02)" fill="#ff9aa2"/><path d="M197.15,104.4a3.6,3.6,0,0,0,.32-.69c0-.94-18.88-32.71-50.75-32.71S96,102.77,96,103.71a3,3,0,0,0,.31.69c2.52-3.51,20.89-23.71,50.41-23.71S194.62,100.89,197.15,104.4Z" transform="translate(-0.12 -0.02)" fill="#c689ff"/><path d="M96.29,104.4a41.56,41.56,0,0,0,2.54,4c6.39,9,23.25,28,47.89,28s41.51-19,47.89-28a41.56,41.56,0,0,0,2.54-4c-2.53-3.51-20.9-23.71-50.43-23.71S98.83,100.89,96.29,104.4Z" transform="translate(-0.12 -0.02)" fill="#d9bfff"/><circle cx="146.6" cy="100.05" r="29.07" fill="#7542e5"/><circle cx="146.48" cy="100.05" r="21.8" fill="#9059ff"/><circle cx="146.66" cy="99.99" r="14.6" fill="#45278d"/><circle cx="155.85" cy="89.79" r="7.84" fill="#fff"/><path d="M194.61,108.45a18.94,18.94,0,0,0,2.86-3.46,2.26,2.26,0,0,0-.32-.54A41.56,41.56,0,0,1,194.61,108.45Z" transform="translate(-0.12 -0.02)" fill="#d9bfff"/></svg>
```
Comment (maxxcrawford) [inline public/img/svg/email-relay-touts.svg]: Do these SVGs need the Mozilla Privacy license comment? It appears on in the Notification SVG `/public/img/svg/notification.svg`: 

```
<!-- This Source Code Form is subject to the terms of the Mozilla Public
   - License, v. 2.0. If a copy of the MPL was not distributed with this
   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
```

File: views/partials/dashboards/breaches-dash.hbs [hunk 6]
```diff
@@ -37,9 +37,24 @@
...
+      {{#eachFromTo this.verifiedEmails 0 1 }}
         {{> email-card }}
-      {{/each}}
+      {{/eachFromTo}}
+      {{#if ./../supportedLocalesIncludeEnglish}}
+        <div class="private-relay-dashboard-card">
+          <a href="/protect-my-email" class="private-relay-dashboard-card-border flx private-relay-cta" data-analytics-label="join-the-waitlist-link" data-user-is-signed-up={{ ./../userHasSignedUpForRelay }}>
+            <div class="private-relay-dashboard-card-content flx">
+              <div class="private-relay-dashboard-card-copy">
+                <p class="bold ff-Met private-relay-dashboard-card-subhead">Clear inbox clutter. Keep your personal email private.</p>
```
Comment (maxxcrawford) [inline views/partials/dashboards/breaches-dash.hbs]: Is it standard practice for the default (EN-US) strings to already be in the element, or should they be placed dynamically from the localization render?

Comment (lesleyjanenorton) [reply views/partials/dashboards/breaches-dash.hbs]: Definitely not standard practice but these strings are not being localized unfortunately and will be removed once the test is over anyway.

File: public/img/svg/email-relay-touts.svg (hunk 5 above)
Comment (lesleyjanenorton) [reply public/img/svg/email-relay-touts.svg]: Good question! It doesn't appear on most of our SVGs but I will ask legal. I think we're ok to proceed without it for now and add in later if necessary.

**Pull request merged**